wire-ios-data-model/Source/Model/UserClient/E2eIdentityProvider.swift

```diff
@@ -0,0 +1,59 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+public struct E2eIdentityCertificate {
+    public var certificateDetails: String
+    public var expiryDate: Date
+    public var certificateStatus: String
```

Comment:
We can have an enum with all possible certificate statuses

Reply:
`E2EIdentityCertificateStatus` Should I move it to WireDataModel and import this everywhere it is used?

Reply:
Yes, I would have it as strict type.

Reply:
Will do this part of the next PR.